Assets/Scripts/UI/LevelSelectManager.cs

```diff
@@ -0,0 +1,134 @@
+
+
+    private List<GameObject> instantiatedCards = new List<GameObject>();
+    private InputManager input;
+    private EventSystem eventSystem;
+
+    public void Start()
+    {
+        eventSystem = EventSystem.current;
+        
+        float angleBetween = totalDegrees / ((float)levelCards.Count + 1);
+        float startAngle = 180 - totalDegrees / 2;
+        
+        for (int i = 0; i < levelCards.Count; i++)
+        {
+            Vector3 offsetPosition = new Vector3(0f, 0f, 3f);
+            Quaternion rotation = Quaternion.Euler(0f, startAngle + angleBetween * (i + 1), 0f);
+
+            //Create new card parent to rotate independently of parent
+            GameObject newParent = new GameObject(levelCards[i].name + "Parent");
+            newParent.transform.SetParent(parent.transform);
+            newParent.transform.position = parent.transform.position;
+            newParent.transform.localScale = Vector3.one;
+            
+            //Instantiate levelcard and attatch to cardparent
+            GameObject levelCard = Instantiate(levelCards[i].getLevelCard(), newParent.transform.position, Quaternion.Euler(-90f, 180f, 180f), newParent.transform);
+            levelCard.transform.localPosition += new Vector3(newParent.transform.localPosition.x, newParent.transform.localPosition.y, newParent.transform.localPosition.z - 0.8f);
+            newParent.transform.rotation = rotation;
+
+            //Add levelcard to list for later use
+            instantiatedCards.Add(levelCard);
+        }
+
+        if (startButton != null && mainMenuController != null)
+        {
+            startButton.onClick.AddListener(HandleStartClick);
+
+            Navigation backNavigation = backButton.navigation;
+            backNavigation.mode = Navigation.Mode.Explicit;
+            UnityEngine.UI.Button firstButton = instantiatedCards[0].GetComponent<UnityEngine.UI.Button>();
+
+            if (firstButton != null)
+            {
+                backNavigation.selectOnUp = firstButton;
+            }
+
+            backButton.navigation = backNavigation;
+        }
+
+        for (int i = 0; i < levelCards.Count; i++)
+        {
+            UnityEngine.UI.Button currentButton = instantiatedCards[i].GetComponent<UnityEngine.UI.Button>();
+            if (currentButton != null)
+            {
+                SetupButtonNavigation(currentButton, i, levelCards.Count);
+            }
+        }
+    }
+
+    public void SetPlayerInput(InputManager input)
+    {
+        this.input = input;
+        input.onSelect += HandleCardClick;
+    }
+
+    private void SetupButtonNavigation(UnityEngine.UI.Button button, int index, int totalCount)
+    {
+        Navigation navigation = button.navigation;
+        navigation.mode = Navigation.Mode.Explicit;
+
+        if (index == 0 && totalCount > 1) //First levelcard button
+        {
+            navigation.selectOnRight = instantiatedCards[index + 1].GetComponent<UnityEngine.UI.Button>();
```

Comment:
I'd prefer wraparound here (when on one end you end up at the other when you try to go "beyond"). We've got that in all other menus now I think? This also goes for the vertical movement to the back button. With this in mind, you should be able to use `(index +/- 1).Mod(totalCount)` as index in the `instantiatedCards` array, which also makes the if branches unnecessary.

Reply:
Fixed wraparound